kroxylicious-systemtests/src/main/java/io/kroxylicious/systemtests/templates/kroxylicious/KroxyliciousSecretTemplates.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.systemtests.templates.kroxylicious;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Base64;
+
+import io.fabric8.kubernetes.api.model.SecretBuilder;
+
+/**
+ * The type Kroxylicious secret templates.
+ */
+public class KroxyliciousSecretTemplates {
+
+    /**
+     * Create registry credentials secret.
+     *
+     * @param configFolder the config folder
+     * @param namespace the namespace
+     * @return the secret builder
+     */
+    public static SecretBuilder createRegistryCredentialsSecret(String configFolder, String namespace) {
```

Comment:
configFolder is misnamed.  It is not the name of a folder, it is the name of a file.

Reply:
It seems a bit weird to return the builder.  I'd just return the Secret and document the fact that the rv is null in the case where there is no secret to create.

Reply:
done

Reply:
> It seems a bit weird to return the builder. I'd just return the Secret and document the fact that the rv is null in the case where there is no secret to create.

you didn't do anything with this comment.

Reply:
> It seems a bit weird to return the builder. I'd just return the Secret and document the fact that the rv is null in the case where there is no secret to create.

All the templates in the framework return a builder, for that reason I decided to do it in the same way.